draft-ietf-privacypass-key-consistency.md

```diff
@@ -91,59 +99,97 @@ solutions to this problem.
+
+Key Set Identifier (Set ID):
+: A unique identifier for a key set.
+
+Client:
+: An entity that uses a key in a reliant system.
 
-The KCCS's consistency model is dependent on the implementation and reliant system's threat model.
+Source:
+: An entity that provides key material for use by clients.
 
-# Core Requirements {#reqs}
+The key consistency model is dependent on the implementation and reliant system's threat model.
+
+# Consistency Requirements {#reqs}
 
 Privacy-focused protocols which rely on widely shared public keys typically
-require keys be consistent and correct. Informally, key consistency is the
-requirement that all users who communicate with an entity share the same view
-of the key associated with that entity; key correctness is that the key's
-secret information is controlled by the intended entity and is not known to be
-available to an external attacker.
-
-Some protocols depend on large sets of users with consistent keys for privacy
-reasons. Specifically, all users with a consistent key represent an anonymity
-set wherein each user of the key in that set is indistinguishable from the
-rest. An attacker that can actively cause inconsistent views of keys can
-therefore compromise user privacy.
-
-An attacker that can cause a user to use an incorrect key will likely compromise
-the entire protocol, not just privacy.
-
-Reliant systems must also consider agility when trying to satisfy these requirements. A naive solution to
-ensuring consistent and correct keys is to only use a single, fixed key pair for the entirety of
-the system. Users can then embed this key into software or elsewhere as needed, without any additional
-mechanics or controls to ensure that other users have a different key. However, this solution clearly
+require keys be consistent. Informally, key consistency is the
+requirement that all clients who use a source-provided key in some reliant system
+share the same view of the key. Some protocols depend on large sets of clients
+with consistent keys for privacy reasons. Specifically, all clients with a
+consistent key represent an anonymity set wherein each client of the key in
+that set is indistinguishable from the rest. An attacker that can actively
+cause inconsistent views of keys can therefore compromise client privacy.
+
+## Consistency Definitions
+
+Formally, consistency is a predicate defined based on key sets. Typically, clients
+try to assess consistency of one key against one or more keys, but there are
+no restrictions on whether the clients holding those keys are the same.
+
+There are two different predicates for consistency, defined below.
+
+- Consistency: Two key sets with the same set ID are consistent if and only if (iff) the intersection
+  of key IDs for each set is non-empty.
+- Global consistency: A key set is globally consistent iff, for all key sets with the
+  same set ID, the intersection between each set of key IDs and each other set is non-empty.
```

Comment:
I'm not 100% on this definition.  Two concerns:

1. This focuses on identifiers, but we know that one identifier can reasonably identify multiple keys.
2. The intersection text is strange.  If keys sets overlap, that doesn't guarantee that users of those key sets will choose to exercise overlapping portions.  (You talk about this below as having unions that are consistent, but this is not the thing I would seek to emphasize.)

Reply:
I think the intersection point is good. As a counter example, consider the following:

- Cache gets key set {K1, K2}
- Client C1 gets key K1
- Client C2 gets key K2
- Client C1 and C2 would conclude consistent if they checked against the cache, but they're using inconsistent keys

I'm going to make this emphasize exact equality instead.